### PR TITLE
test: Add 5 multi-threading test cases for ReadWritePolicyManager

### PR DESCRIPTION
**Pull request by sociometry**
test: Add 5 multi-threading test cases for ReadWritePolicyManager

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sociometry plz fix:

image

**Comment by hsluoyz** [comment]
@sagilio

File: Casbin.UnitTest/MultithreadingTests/MultithreadingTests.cs [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+            _waitWhenWrite = waitWhenWrite;
+            _waitWhenRead = waitWhenRead;
+        }
+
+        public static IPolicyManager Create(List<string> result, bool waitWhenWrite, bool waitWhenRead)
+        {
+            return new ReaderWriterPolicyManagerForTest(DefaultPolicyStore.Create(), result, waitWhenWrite, waitWhenRead);
+        }
+
+        public override Task<bool> RemovePolicyAsync(string section, string policyType, IEnumerable<string> rule)
```
Comment (sagilio) [inline Casbin.UnitTest/MultithreadingTests/MultithreadingTests.cs:30]: It overwrites all content, we may not yet get the error even though the base is broken, can we mock an adapter to test?

**Review by sagilio — CHANGES_REQUESTED** [review]
Hope to add tests without lock timeouts

**Comment by sagilio** [comment]
ReadWritePolicyManager has been removed at #282

**Pull request closed without merging**